test/langtools/jdk/javadoc/lib/javadoc/tester/JavadocTester.java

```diff
@@ -762,56 +768,55 @@ public void checkUnique(String path, String... strings) {
+        Path bd1 = Path.of(baseDir1);
+        Path bd2 = Path.of(baseDir2);
         for (String file : files) {
-            diff(bd1, bd2, file);
+            diff(bd1, bd2, Path.of(file));
         }
     }
 
     /**
-     * Copies a directory from one place to another.
+     * Copies the contents of a directory to another directory.
      *
-     * @param targetDir the directory to copy.
-     * @param destDir the destination to copy the directory to.
+     * @param fromDir the directory containing the files to be copied
+     * @param toDir   the destination to which to copy the files
      */
-    // TODO: convert to using java.nio.Files.walkFileTree
-    public void copyDir(String targetDir, String destDir) {
-        try {
-            File targetDirObj = new File(targetDir);
-            File destDirParentObj = new File(destDir);
-            File destDirObj = new File(destDirParentObj, targetDirObj.getName());
-            if (! destDirParentObj.exists()) {
-                destDirParentObj.mkdir();
-            }
-            if (! destDirObj.exists()) {
-                destDirObj.mkdir();
-            }
-            String[] files = targetDirObj.list();
-            for (String file : files) {
-                File srcFile = new File(targetDirObj, file);
-                File destFile = new File(destDirObj, file);
-                if (srcFile.isFile()) {
-                    out.println("Copying " + srcFile + " to " + destFile);
-                    copyFile(destFile, srcFile);
-                } else if(srcFile.isDirectory()) {
-                    copyDir(srcFile.getAbsolutePath(), destDirObj.getAbsolutePath());
-                }
-            }
-        } catch (IOException exc) {
-            throw new Error("Could not copy " + targetDir + " to " + destDir);
-        }
+    public void copyDir(String fromDir, String toDir) {
+        copyDir(Path.of(fromDir), Path.of(toDir));
     }
 
     /**
-     * Copies a file.
+     * Copies the contents of a directory to another directory.
      *
-     * @param destfile the destination file
-     * @param srcfile the source file
-     * @throws IOException
+     * @param fromDir the directory containing the files to be copied
+     * @param toDir   the destination to which to copy the files
      */
-    public void copyFile(File destfile, File srcfile) throws IOException {
-        Files.copy(srcfile.toPath(), destfile.toPath());
+    public void copyDir(Path fromDir, Path toDir) {
+        out.println("Copying " + fromDir + " to " + toDir);
+        try {
+            Files.createDirectories(toDir);
+            Files.walkFileTree(fromDir, new SimpleFileVisitor<>() {
+                @Override
+                public FileVisitResult preVisitDirectory(Path fromSubdir, BasicFileAttributes attrs) throws IOException {
+                    Path toSubdir = toDir.resolve(fromDir.relativize(fromSubdir));
+                    if (Files.exists(toSubdir)) {
```

Comment:
Shouldn't this check be negated? Otherwise, sub-folders are not copied. On the second thought, why do we need this check at all?

Reply:
Oops.
Whether it is needed depends on whether you think the target directory exists or not.  In the one existing use case, the directory should not exist and so the  check is not necessary.   Given that tests run in an empty scratch dir, we could reasonably assume that is the case.

Reply:
This method (`copyDir`) does not really belong in `JavadocTester`.  It's only used in one test, so could be moved there, but a better home is in `ToolBox` which already has similar methods.  I'll move it there.

Separately, I see `ToolBox` could use some TLC/updates ... e.g. use `Path.of` instead of `Paths.get` but I'll defer that to a separate issue.